### ISSUE TITLE
Cacheing Loaded imports too early, resulting in promises being resolved before import has loaded.

### ISSUE THREAD

**Issue by akc42** [Polymer/lazy-imports#30]
I have a situation where I am calling importLazyGroup twice for the same group (but more importantly for the same import hrefs) in quick sucession.  The first kicks off the importHref call to fetch the resource.  The second call to importLazyGroup comes before the first completes.  However it completes immediately, resulting in the caller being told it is loaded when it isn't yet.

The reason for this is the code below taken from `lazy-imports-import.html`
```
          var resolved = callingElement.resolveUrl(href);
          if (!__loadedImports[resolved]) {
            __loadedImports[resolved] = true;
            var importHref = Polymer.importHref || callingElement.importHref;
            importHref(resolved, onLoad, onFailure);
          } else {
            resolve({loaded: href});
          }

```
On the first call through for a given href you add it to the `_loadedImports` cache as true **before** the `importHref` is called (and *especially* before it completes).

On the second call through the href matches the cache as `true`, even though it might not yet have completed and so the href is resolve immediately.

**Comment by stramel** [comment]
I believe this is the edge case @justinfagnani was referring to in #28 and once we get #29 in, it should resolve this issue. Thanks for reporting!

**Comment by stramel** [comment]
@akc42 @justinfagnani Do one of you two mind closing this? Should have been addressed with #36

**Comment by justinfagnani** [comment]
Done :)
